Add OpenAI-CLIP-Image-Text-Embeddings-ViT-Large-Patch14-336 Embeddings model (#2007)
Azure/azureml-assets: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md

> The `CLIP` model was developed by OpenAI researchers to learn about what contributes to robustness in computer vision tasks and to test the ability of models to generalize to arbitrary image classification tasks in a zero-shot manner. The model uses a ViT-L/14 Transformer architecture trained at 336x336 pixel resolution as an image encoder and uses a masked self-attention Transformer as a text encoder. The model was trained on publicly available image-caption data, which was gathered in a mostly non-interventionist manner. The model is intended as a research output for research communities, and the primary intended users of these models are AI researchers. The model has been evaluated on a wide range of benchmarks across a variety of computer vision datasets, but it currently struggles with respect to certain tasks such as fine-grained classification and counting objects. The model also poses issues with regards to fairness and bias, and the specific biases it exhibits can depend significantly on class design and the choices one makes for categories to include and exclude.

Reviewer: please align description as this [doc](https://microsoftapc-my.sharepoint.com/:w:/r/personal/ayushmishra_microsoft_com/_layouts/15/doc2.aspx?sourcedoc=%7B2E13DB38-A1A9-4148-B1DC-AC14A76E1F72%7D&file=System%20Registry%20model%20card%20alignment.docx&nav=eyJjIjo0MTgzNzU3OX0&action=default&mobileredirect=true&share=IQE42xMuqaFIQbHcrBSnbh9yAamNOJxrsZ9DvamUluUGp40&wdOrigin=TEAMS-ELECTRON.p2p_ns.bim&wdExp=TEAMS-CONTROL&wdhostclicktime=1703143808964&web=1).

Also you can refer to this [PR] (https://github.com/Azure/azureml-assets/pull/2010)

### Inference samples

Inference type|Python sample (Notebook)|CLI with YAML
|--|--|--|
Real time|<a href="https://aka.ms/azureml-infer-sdk-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-cli-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.sh</a>
Batch|<a href="https://aka.ms/azureml-infer-batch-sdk-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-batch-cli-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.sh</a>

### Sample inputs and outputs (for real-time inference)

#### Sample input for image embeddings

```json
{
   "input_data":{
      "columns":[
         "image", "text"
      ],
      "index":[0, 1],
      "data":[
         ["image1", ""],
         ["image2", ""]
      ]
   }
}
```
Note: "image1" and "image2" should be publicly accessible urls or strings in `base64` format

#### Sample output

```json
[
    {
        "image_features": [-0.92, -0.13, 0.02, ... , 0.13],
    },
    {
        "image_features": [0.54, -0.83, 0.13, ... , 0.26],
    }
]
```

> Note: returned embeddings have dimension 768 and are not normalized

Reviewer 2: 768 -> 1024

#### Sample input for text embeddings

```json
{
   "input_data":{
      "columns":[
         "image", "text"
      ],
      "index":[0, 1],
      "data":[
         ["", "sample text 1"],
         ["", "sample text 2"]
      ]
   }
}
```

#### Sample output

```json
[
    {
        "text_features": [0.42, -0.13, -0.92, ... , 0.63],
    },
    {
        "text_features": [-0.14, 0.93, -0.15, ... , 0.66],
    }
]
```

> Note: returned embeddings have dimension 768 and are not normalized

Reviewer 2: 768 -> 1024

#### Sample input for image and text embeddings

```json
{
   "input_data":{
      "columns":[
         "image", "text"
      ],
      "index":[0, 1],
      "data":[
         ["image1", "sample text 1"],
         ["image2", "sample text 2"]
      ]
   }
}
```
Note: "image1" and "image2" should be publicly accessible urls or strings in `base64` format

#### Sample output

```json
[
    {
        "image_features": [0.92, -0.13, 0.02, ... , -0.13],
        "text_features": [0.42, 0.13, -0.92, ... , -0.63]
    },
    {
        "image_features": [-0.54, -0.83, 0.13, ... , -0.26],
        "text_features": [-0.14, -0.93, 0.15, ... , 0.66]
    }
]
```

> Note: returned embeddings have dimension 768 and are not normalized

Reviewer 2: 768 -> 1024

Author: <img width="820" alt="image" src="https://github.com/Azure/azureml-assets/assets/38021489/f5e9b29d-1fc0-42af-be15-0157322c8877">
The embedding dimension is 768. I have also verified this by inspecting the embeddings returned from the endpoint.

# License
The MIT License is a permissive free software license originating at the Massachusetts Institute of Technology (MIT). The license allows users to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the software, subject to the conditions that the copyright notice and permission notice appear in all copies or substantial portions of the software.

> ### Inference samples

Reviewer 3: ```suggestion
# Inference samples
```

Inference type|Python sample (Notebook)|CLI with YAML
|--|--|--|
Real time|<a href="https://aka.ms/azureml-infer-sdk-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-cli-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.sh</a>
Batch|<a href="https://aka.ms/azureml-infer-batch-sdk-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-batch-cli-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.sh</a>

> 

Reviewer 3: For Sample I/O, format that's followed else where:

```
# Sample input and output

### Sample input

### Sample output

#### Visualization of inference result for a sample image
```

> ### Sample inputs and outputs (for real-time inference)

Reviewer 3: ```suggestion
# Sample input and output (for real-time inference)
```

Reviewer 4: Please remove: `(for real-time inference)` from the end as well

> #### Sample input for image embeddings

Reviewer 3: ```suggestion
### Sample input for image embeddings
```

Reviewer 3: We should perhaps add a section for text embeddings for completeness.

Reviewer 3: NVM, its there.

> #### Sample input for image embeddings

Reviewer 3: ```suggestion
### Sample input for image embeddings
```

```json
{
   "input_data":{
      "columns":[
         "image", "text"
      ],
      "index":[0, 1],
      "data":[
         ["image1", ""],
         ["image2", ""]
      ]
   }
}
```
Note: "image1" and "image2" should be publicly accessible urls or strings in `base64` format

> #### Sample output

Reviewer 3: ```suggestion
### Sample output
```

```json
[
    {
        "image_features": [-0.92, -0.13, 0.02, ... , 0.13],
    },
    {
        "image_features": [0.54, -0.83, 0.13, ... , 0.26],
    }
]
```
Note: returned embeddings have dimension 768 and are not normalized

> #### Sample input for text embeddings

Reviewer 3: ```suggestion
### Sample input for text embeddings
```

```json
{
   "input_data":{
      "columns":[
         "image", "text"
      ],
      "index":[0, 1],
      "data":[
         ["", "sample text 1"],
         ["", "sample text 2"]
      ]
   }
}
```

> #### Sample output

Reviewer 3: ```suggestion
### Sample output for text embeddings
```

```json
[
    {
        "text_features": [0.42, -0.13, -0.92, ... , 0.63],
    },
    {
        "text_features": [-0.14, 0.93, -0.15, ... , 0.66],
    }
]
```
Note: returned embeddings have dimension 768 and are not normalized

> #### Sample input for image and text embeddings

Reviewer 3: ```suggestion
### Sample input for image and text embeddings
```

```json
{
   "input_data":{
      "columns":[
         "image", "text"
      ],
      "index":[0, 1],
      "data":[
         ["image1", "sample text 1"],
         ["image2", "sample text 2"]
      ]
   }
}
```
Note: "image1" and "image2" should be publicly accessible urls or strings in `base64` format

> #### Sample output

Reviewer 3: ```suggestion
#### Sample output for image and text embeddings
```
